### PR TITLE
Add print-state function, and make example code in custom-handler* docstring compile

### PR DESCRIPTION
**Pull request by kenrestivo**
Add a print-state function to show state of routes and middleware.

Also, fix example code given in the docstring for custom-handler\* so that it compiles.

### CODE REVIEW

File: src/noir/validation.clj [hunk 1]
```diff
@@ -43,8 +43,9 @@
 (defn valid-file?
   "Returns true if a valid file was supplied"
   [m]
-  (and (> (:size m) 0)
-       (:filename m)))
+  (and (:size m)
+   (> (:size m) 0)
```
Comment (Raynes) [inline src/noir/validation.clj:47]: This indentation is wrong. Arguments should be lined up.

**Comment by Raynes** [comment]
I'm a little confused. This is like 2-3 different pull requests stuffed together, isn't it? Looks like you may have put everything on the same branch before opening the request. It isn't a big deal, but you explain everything that was done for me? :)

**Comment by kenrestivo** [comment]
Sure.

validation.clj (5fa8f92), adds a check to make sure that there's a :size key. If the form was submitted incorrectly (not using multipart), there will be no :size key. Rather than throw an exception, this will just invalidate the field. The indentation is whatever clojure-mode did; I can change it if you like.

test/util.clj (106b6c0, 0c7cce1) adds lines to display the memory store, wrappers, and optionally status pages, and fixes indentation.

session.clj (cfbb62e, 514874f, e4062ab, 6fca574, fdfadec ) adds some machinery to noir-session to make the session emulate being pure. Some ring libraries like friend need to dissoc keys from the session functionally, without them getting cached and then re-added by the _noir-session_ dynamic var. In order to do that, the patch segregates the _noir-session_ managed keys (via get! put! remove!) into a  :noir map inside the session. I should caution that this may or may not be a breaking change; if someone depended upon :session keys saved persistently in a non-memory store, they will have moved once they restart the app with this patch applied. The changes to test/core.clj add regression tests for the patches to session.clj.

119ee0a is  lein2-compatibility (/target/ to .gitignore)

486b173 is just a doc fix in compojure-route.

**Comment by kenrestivo** [comment]
I'm going to bust this out into several pull requests and feature branches.

**Pull request closed without merging**